src/main.rs

```diff
@@ -124,14 +137,144 @@ fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Erro
     ))
 }
 
+// If there are no errors, this returns the computed diff of the target spec's
+// current branch and base branch (master or main). The output should be
+// filtered by `sanitized_diff_lines()`.
+fn git_diff(path: &Path) -> Result<String, clap::error::Error> {
+    // Extract the filename itself, as well as the directory from `path`.
+    assert!(path.is_file());
+    let filename_without_path = path.file_name().unwrap().to_str().unwrap();
+    let directory = path.parent().unwrap().to_str().unwrap();
+
+    // Get the name of the git branch that the spec is currently on.
+    let current_branch = std::process::Command::new("git")
+        .arg("-C")
+        .arg(directory)
+        .arg("branch")
+        .arg("--show-current")
+        .output()
+        .expect("Failed to run `git branch --show-current`");
+    let current_branch = String::from_utf8(current_branch.stdout).unwrap();
+    let current_branch = current_branch.trim();
+
+    // Get the base branch to compare `current_branch` to with in `git diff`. We
+    // expect it to be either `master` or `main`, and fail otherwise.
+    let branches = std::process::Command::new("git")
+        .arg("-C")
+        .arg(directory)
+        .arg("for-each-ref")
+        .arg("--format=%(refname:short)")
+        .output()
+        .expect("Failed to find the base branch to compare current branch '${}' with");
+    let branches = String::from_utf8(branches.stdout).unwrap();
+    let branches = branches.split('\n');
```

Comment:
I keep running into this situation where I want to call something like split() or trim() on the output of a function that produces an owned type like `String`, and I have to do them on separate lines. If I were to try and combine these two lines into:

```rust
    let branches = String::from_utf8(branches.stdout).unwrap().split('\n');
```

Then I get a compiler error about how the `String` returned from `from_utf8` goes out of scope just after `split()` grabs slices over it, so we get a lifetime issue. Having two distinct lines for this seems fine, but I can't tell if there is a better or more-idiomatic way of doing this.